repo/blob/gcs/gcs_storage.go

```diff
@@ -273,38 +251,28 @@ func tokenSourceFromCredentialsJSON(ctx context.Context, data json.RawMessage, s
 func New(ctx context.Context, opt *Options, isCreate bool) (blob.Storage, error) {
 	_ = isCreate
 
-	var ts oauth2.TokenSource
-
-	var err error
+	if opt.BucketName == "" {
+		return nil, errors.New("bucket name must be specified")
+	}
 
 	scope := gcsclient.ScopeReadWrite
 	if opt.ReadOnly {
 		scope = gcsclient.ScopeReadOnly
 	}
 
-	if sa := opt.ServiceAccountCredentialJSON; len(sa) > 0 {
-		ts, err = tokenSourceFromCredentialsJSON(ctx, sa, scope)
-	} else if sa := opt.ServiceAccountCredentialsFile; sa != "" {
-		ts, err = tokenSourceFromCredentialsFile(ctx, sa, scope)
-	} else {
-		ts, err = google.DefaultTokenSource(ctx, scope)
-	}
+	clientOptions := []option.ClientOption{option.WithScopes(scope)}
 
-	if err != nil {
-		return nil, errors.Wrap(err, "unable to initialize token source")
+	if j := opt.ServiceAccountCredentialJSON; len(j) > 0 {
+		clientOptions = append(clientOptions, option.WithCredentialsJSON(j))
+	} else if fn := opt.ServiceAccountCredentialsFile; fn != "" {
+		clientOptions = append(clientOptions, option.WithCredentialsFile(fn))
 	}
```

Comment:
For completeness maybe to add:
```suggestion
	} else {
	        ts, err := google.DefaultTokenSource(ctx)
		if err != nil {
			return nil, errors.Wrap(err, "failed to get default token source")
		}

		clientOptions = append(clientOptions, option.WithTokenSource(ts))
	}
```
since before the 3 flows were supported